conf/report/report-template.html

```diff
@@ -65,13 +65,80 @@
       $('#filter_wrapper').remove();
       iter = 1;
       var div = $('<div id="filter_wrapper"><button id="mainButton" onclick="addOptions()">Add</button></div>').insertAfter('#headline');
-      filter.length = 0;
+      filters.length = 0;
     }
 
     function applyFilter() {
       apply_active = true;
-      if(!document.getElementById('removeButton'));
+      if(!document.getElementById('removeButton'))
         var remove_all = $('<button id="removeButton" onclick="removeAll()">Remove all filters</button>').insertAfter('#applyButton');
+      
+      report = $('#report_table').DataTable();
```

Comment:
When I tried to declare this as a global variable, i.e., outside of all the functions - it was showing null

Reply:
That's because your code runs before the table is created. The table is created below in `$(document).ready( function () {...}`.

Reply:
That's because your code runs before the table is created. This happens in a callback created in line 279 `$(document).ready(function () {...})`.